core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -143,6 +143,12 @@ class HistoryServer(
     appCache.stop()
   }
 
+  // For testing - override stop timeout used by jetty
+  private[history] def setStopTimeout(timeout: Long): Unit = {
```

Comment:
this was 30 secs additional for just the one test i added. if more such tests are added the time would add up - so i felt the option to bypass the wait was useful to have. assuming more of the UI could transition to ajax etc I could move this utility function up into the base `WebUI` class so it's available more generally for future UI tests - not just history server?